### PR TITLE
project token auth

### PR DESCRIPTION
**Pull request by dheniges**
So you have a project token? And you want to use it to access project resources? Now you can!!!

- custom passport strategy to use project tokens
- code path in permissions is pretty quick to skip over orgs, and project user roles when using project tokens, so little overhead there
- The most overhead is having to run the jwt auth and fail before running the custom project token jwt auth. Might be a way to short-circuit that down the road

### CODE REVIEW

File: api/src/businesstime/project.test-i.js [hunk 1]
```diff
@@ -112,6 +112,35 @@ describe('BusinessProject', () => {
...
+      describe('when the target project has the passed in orgId', () => {
+        beforeAll(async () => {
+          targetProjectId = factoryProjects[0].get('id')
+          project = await BusinessProject.findById(targetProjectId, constants.ORG_ID)
+        })
+
+        it('should return a project as POJO', () => {
+          expect(project.id).toBe(targetProjectId)
+          expect(project.constructor).toBe(Object)
+          expect(Object.keys(project)).toEqual(expect.arrayContaining(PUBLIC_FIELDS))
```
Comment (jjidt) [inline api/src/businesstime/project.test-i.js]: should we also check for at least one non-public field?

File: api/src/businesstime/projecttoken.test-i.js [hunk 2]
```diff
@@ -119,6 +119,27 @@ describe('BusinessProjectToken', () => {
...
+      factoryToken = (await TokenFactory.createMany(1, {
+        projectId: 123,
+        orgId: constants.ORG_ID
+      }))[0]
+
+      token = await BusinessProjectToken.findByIdUnsanitized(factoryToken.id, factoryToken.orgId)
+    })
+
+    it('should return public fields', () => {
+      expect(Object.keys(token)).toEqual(expect.arrayContaining(PUBLIC_FIELDS))
```
Comment (jjidt) [inline api/src/businesstime/projecttoken.test-i.js:135]: same here, do we want to have a check for a non-public field

**Review by jjidt — COMMENTED** [review]
manually tested with multiple tiers and it works great! great tests, just one suggestion

File: api/src/businesstime/projecttoken.test-i.js (hunk 2 above)
Comment (dheniges) [reply api/src/businesstime/projecttoken.test-i.js:135]: the next test checks for the secret

File: api/src/businesstime/project.test-i.js (hunk 1 above)
Comment (dheniges) [reply api/src/businesstime/project.test-i.js]: I accidentally copied and didn't remove these tests, put them in the wrong file. Deleted them now.

**Review by jjidt — APPROVED** [review]

**Pull request merged**